plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/ProbabilisticSampler.java

```diff
@@ -67,7 +56,14 @@ public SamplingResult shouldSample(
         Attributes attributes,
         List<LinkData> parentLinks
     ) {
-        return getSampler().shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
+        double newSamplingRatio = telemetrySettings.getSamplingProbability();
```

Comment:
Please revert this changes, those are not related

Reply:
Earlier we were using [ProbabilisticSampler.java](https://github.com/opensearch-project/OpenSearch/pull/10136/files#diff-2bbe60776710b615ecb0bddea8805f4adb6ec7e3620a7bae455abebcf3790506) as a common place where all samplers like action & probabilistic sampler were added, as a result we added getSampler() method tp identify the best sampler based on the requirement. 

But now those changes are not need as we have different sampler for different purpose.

Reply:
What functionally changed in this code? I see that getSampler() method was inlined as-is here

Reply:
The getSampler() would return either an actionBasedSampler or Probability sampler based on the need.